include/glow/IR/IR.h

```diff
@@ -225,6 +225,60 @@ class Instruction : public Value, public TaggedListNode<Instruction> {
   void dumpOperands(llvm::raw_ostream &os) const;
 };
 
+/// Different stages of processing an IR instruction. Transitions between stages
+/// always happen in the same order as they are defined below, i.e.
+/// PREPROCESSING -> PROCESSING -> POSTPROCESSING -> DONE.
+/// In particular, there is no possibility for any transition in the backwards
+/// direction.
+enum class IRInstructionProcessingStage {
+  /// Instruction is going to be processed.
+  PREPROCESSING,
+  /// Instruction is being processed.
+  PROCESSING,
+  /// Instruction was just processed.
+  POSTPROCESSING,
+  /// Instruction processing is done.
+  DONE,
```

Comment:
Do we still need DONE here since the caller manages the stage? I.e. the handler doesn't return a stage so not sure this would ever be used?